### PR TITLE
LPS-130919 Move configure Session Property Vocabularies from System Settings to Instance Settings 

### PR DESCRIPTION
**Pull request by YurenaC**
Motivation
----------------

Currently, segmentation settings exist only at System level, so there’s no way to configure the feature at the Instance level.
In order to facilitate the cloud-based products development, we need to make it available at the instance level.

[Link to story or ticket](https://issues.liferay.com/browse/LPS-130919)

Proposed Solution
----------------

To easily solve this problem, the scope of the Session Property Vocabularies has been set to COMPANY, so after a few simple changes to allow multiple assetVocabularies, with different entityFields assigned depending on its level (system or instance) it works as expected

Steps to Verify:
----------------

1. Start the portal and go to "Control Panel > System Settings > In Content and Data select Segments"
2. In "Virtual Instance Scope" select "Session Property Vocabularies"
3. Select "add" and choose "browser" in "Session Property Name" and "Audience" in Vocabulary Name, click Save button
4. Check point: Now you can see "browser" listed in "Configuration Entries"
5. Go to "Control Panel > Instance Settings > In Content and Data select Segments"
6. In "Virtual Instance Scope" select "Session Property Vocabularies"
7. Select "add" and choose "browser" in "Session Property Name" and "Stage" in Vocabulary Name, click Save button
8. All should work as expected.
9. Repeat step 7 and a Duplication Error message should be shown.

Change log
----------------

- LPS-130919 Update SegmentsContextVocabularyConfiguration scope to Company
- LPS-130919 Allow overwrite session properties (defined at system level) to a different vocabulary at instance level
- LPS-130919 LPS-130919 Making Session Property Vocabularies available at system scope and also to virtual instance scope in system settings
- LPS-130919 Add SegmentsContextVocabularySegmentsFieldCustomizer
-[LPS-130919 Integration tests

### CODE REVIEW

**Comment by YurenaC** [comment]
ci:test:sf

**Comment by YurenaC** [comment]
ci:test:sf

**Comment by YurenaC** [comment]
Hi @cgoncas,
I created the requested test as unit tests, please, have a look when you have some spare time
Thanks!!

A new PR was created as the previous one was accidentally merged with Tango master (already reverted)

**Comment by YurenaC** [comment]
ci:test:relevant

**Comment by YurenaC** [comment]
ci:test:tango

File: modules/apps/segments/segments-context-vocabulary/src/main/java/com/liferay/segments/context/vocabulary/internal/configuration/persistence/listener/SegmentsContextVocabularyConfigurationModelListener.java [hunk 1]
```diff
@@ -56,16 +60,19 @@ public void onBeforeSave(String pid, Dictionary<String, Object> properties)
...
+		if (_isDefined(
+				String.valueOf(properties.get("assetVocabulary")), entityField,
+				String.valueOf(properties.get("companyId")))) {
+
 			throw new DuplicatedSegmentsContextVocabularyConfigurationModelListenerException(
 				ResourceBundleUtil.getString(
 					_getResourceBundle(),
 					"this-field-is-already-linked-to-one-vocabulary"),
-				SegmentsContextVocabularyConfiguration.class, getClass(),
+				SegmentsContextVocabularyCompanyConfiguration.class, getClass(),
```
Comment (cgoncas) [inline modules/apps/segments/segments-context-vocabulary/src/main/java/com/liferay/segments/context/vocabulary/internal/configuration/persistence/listener/SegmentsContextVocabularyConfigurationModelListener.java:75]: Why do we use SegmentsContextVocabularyCompanyConfiguration, I mean could it be duplicated in SegmentsContextVocabularyConfiguration too, right?

**Comment by YurenaC** [comment]
ci:test:sf

**Comment by YurenaC** [comment]
ci:test:relevant

**Comment by YurenaC** [comment]
ci:test:sf

**Comment by YurenaC** [comment]
ci:test:relevant

**Comment by YurenaC** [comment]
ci:test:relevant

**Pull request closed without merging**